wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/WireRun.kt

```diff
@@ -280,19 +280,24 @@ class WireRun(
           profileLoader = schemaLoader,
         )
 
-        visitors.forEach {
-          it.schemaHandlerStart(handler::class.qualifiedName!!, targetToEmittingRules.getValue(target))
+        eventListeners.forEach {
+          it.schemaHandlerStart(handler, targetToEmittingRules.getValue(target))
         }
         handler.handle(partition.schema, context)
-        visitors.forEach {
+        eventListeners.forEach {
           // TODO(Benoit) Pass the definitions claimed by this target?
-          it.schemaHandlerEnd(handler::class.qualifiedName!!, targetToEmittingRules.getValue(target))
+          it.schemaHandlerEnd(handler, targetToEmittingRules.getValue(target))
         }
       }
     }
-    visitors.forEach(Visitor::schemaHandlersEnd)
+    eventListeners.forEach(EventListener::schemaHandlersEnd)
 
-    errorCollector.throwIfNonEmpty()
+    try {
+      errorCollector.throwIfNonEmpty()
+    } catch (e: SchemaException) {
```

Comment:
Avoid exceptions for flow control here? As is this is ugly

```
val errorsSoFar = errorCollector.errors
if (errorsSoFar.isNotEmpty()) {
  eventListeners.forEach { it.runFailed(errorCollector.errors) }
  throw ...
}
```

Reply:
fixed